crates/task-impls/src/helpers.rs

```diff
@@ -30,3 +38,24 @@ pub async fn broadcast_event<E: Clone + std::fmt::Debug>(event: E, sender: &Send
         }
     }
 }
+/// Calculate the vid disperse information from the payload given a view and membership
+///
+/// # Panics
+/// Panics if the VID calculation fails, this should not happen.
+pub async fn calculate_vid_disperse<TYPES: NodeType>(
+    txns: Vec<u8>,
+    membership: Arc<TYPES::Membership>,
+    view: TYPES::Time,
+) -> VidDisperse<TYPES> {
+    let num_nodes = membership.total_nodes();
+    let vid_disperse = spawn_blocking(move || {
+        #[allow(clippy::panic)]
+        vid_scheme(num_nodes).disperse(&txns).unwrap_or_else(|err|panic!("VID disperse failure:\n\t(num_storage nodes,payload_byte_len)=({num_nodes},{})\n\terror: : {err}", txns.len()))
+    })
+    .await;
```

Comment:
Yes it's actually very important here.  spawn_blocking moves the calculation to a different thread pool than the async executor.  Without it the expensive disperse calculation runs on the async thread pool and blocks the executor.  By doing it this way we free up the async thread to handle other things, even though this one task is still blocked.  Here is a good blog post about it from a tokio core contributor: https://ryhl.io/blog/async-what-is-blocking/ (since my explanation is maybe not the best)

Reply:
Oh, I see -- for some reason, I just assumed something like this was happening implicitly on every sync function call, but I guess you lose some control on execution that way.

Thanks for the link and explanation, very helpful!